packages/react-slider/src/components/Slider/useSlider.tsx

```diff
@@ -0,0 +1,265 @@
+        on(currentTarget, 'pointermove', onPointerMove),
+        on(currentTarget, 'pointerup', onPointerUp),
+        () => currentTarget.releasePointerCapture(pointerId),
+      );
+
+      onPointerMove(ev);
+    },
+    [onPointerMove],
+  );
+
+  const onKeyDown = React.useCallback(
+    (ev: React.KeyboardEvent): void => {
+      const key = getCode(ev);
+
+      if (ev.shiftKey) {
+        if (key === ArrowDownKey || key === ArrowLeftKey) {
+          updateValue(ev, currentValue - step * 10);
+          return;
+        } else if (key === ArrowUpKey || key === ArrowRightKey) {
+          updateValue(ev, currentValue + step * 10);
+          return;
+        }
+      } else if (key === ArrowDownKey || key === ArrowLeftKey) {
+        updateValue(ev, currentValue - step);
+        return;
+      } else if (key === ArrowUpKey || key === ArrowRightKey) {
+        updateValue(ev, currentValue + step);
+        return;
+      } else {
+        switch (getCode(ev)) {
+          case PageDownKey:
+            updateValue(ev, currentValue - step * 10);
+            break;
+          case PageUpKey:
+            updateValue(ev, currentValue + step * 10);
+            break;
+          case HomeKey:
+            updateValue(ev, min);
+            break;
+          case EndKey:
+            updateValue(ev, max);
+            break;
+        }
+      }
+    },
+    [currentValue, max, min, step, updateValue],
+  );
+
+  React.useEffect(() => {
+    if (ref && ref.current) {
+      ref.current.value = currentValue;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref.current.focus = thumbRef?.current?.focus() as any;
+    }
+  }, [currentValue, ref]);
+
+  useMount(() => {
+    if (value !== undefined) {
+      setCurrentValue(clamp(value, min, max));
+    }
+  });
+
+  const valuePercent = getPercent(currentValue, min, max);
+
+  const thumbStyles = {
+    transform: `translateX(${valuePercent}%)`,
+  };
+
+  const trackStyles = { width: `${valuePercent}%` };
+
+  const rootProps: Partial<SliderState> = {
+    className: 'ms-Slider-root',
+    onPointerDown: onPointerDown,
+    onKeyDown: onKeyDown,
+    id: id,
+  };
+
+  const railProps: React.HTMLAttributes<HTMLDivElement> = {
+    className: 'ms-Slider-rail',
+  };
+
+  const trackProps: React.HTMLAttributes<HTMLDivElement> = {
+    className: 'ms-Slider-track',
+    style: trackStyles,
+  };
+
+  const thumbProps: React.HTMLAttributes<HTMLDivElement> & React.RefAttributes<HTMLDivElement> = {
+    className: 'ms-Slider-thumb',
+    ref: thumbRef,
+    tabIndex: 0,
+    role: 'slider',
+    'aria-valuemin': min,
+    'aria-valuemax': max,
+    'aria-valuenow': currentValue,
+    'aria-valuetext': ariaValueText ? ariaValueText(currentValue) : currentValue.toString(),
+    style: thumbStyles,
+  };
+
+  const activeRailProps: React.HTMLAttributes<HTMLDivElement> & React.RefAttributes<HTMLDivElement> = {
+    className: 'ms-Slider-activeRail',
+    ref: railRef,
+  };
+
```

Comment:
We should try to extract a lot of this logic into a `useSliderState` file and just leave the `mergeProps` call here to follow the pattern in other components .